dev/core/src/com/google/gwt/core/ext/linker/impl/StandardLinkerContext.java

```diff
@@ -537,6 +539,10 @@ public void produceOutput(TreeLogger logger, ArtifactSet artifacts,
     }
   }
 
+  public ModuleDef getModule() {
```

Comment:
As we discussed when first working out how to plumb this, I don't really like exposing this here. I don't have a good enough answer to ask you to change it, so just documenting my thoughts in case something strikes one of us (or someone sees this PR in a few years and wants to know why on earth we made this change).

One option is to make a specific getter, `shouldEmbedSourcesInModuleSourcemaps()` sort of thing that just does `return module.isEmbedSourcesContent()`. That would better match getModuleName()/getModuleFunctionName()/getModuleLastModified() - though those three are part of LinkerContext itself. I don't think this should be added to LinkerContext. 

I dislike this option the least. However, it does suggest that we should do the same in CompilerContext - but CompilerContext has a bunch of other calls to getModule() already, so it feels a bit inconsistent.

Another option would be to remove the other three getters, and have all the callers instead call `getModule().getName()` and so on. This provides access to too much of ModuleDef I think, and it isn't immutable, so I really don't care for this. 

---

Separately, I don't love the name `isEmbedSourcesContent` as a method on ModuleDef, nor do I love that the method exists there at all. No other properties exist on ModuleDef itself, so it seems silly to have this one be special enough to be there. That said, it is needed by both the various pre/compile stages and linkers too, so it either needs to be somewhere very accessible, or needs to be duplicated - so I think I'm okay with it staying in ModuleDef _or_ pushing it down into Properties (so you have to call `getModule().getProperties.isBlahBlah()` - that does happen in other places, so it wouldn't be terrible). Renaming it: how about we get SourceMaps in the name somehow - shouldEmbedSourceMapContents() perhaps? sourceMapContentsEmbedEnabled()?